toolchest_client/api/query.py

```diff
@@ -336,12 +335,12 @@ def _upload(self, input_file_paths, input_prefix_mapping, input_is_compressed):
 
     def _upload_docker_image(self, custom_docker_image_id):
         if custom_docker_image_id is None:
-            return
+            return None
         try:  # Try to get the image before creating a repository.
             client = docker.from_env()
             image = client.images.get(custom_docker_image_id)
         except ImageNotFound:
-            raise ToolchestException(f"Unable to find image {custom_docker_image_id}.")
+            return custom_docker_image_id
         except (APIError, DockerException):
             raise EnvironmentError('Unable to connect to Docker. Make sure you have docker installed and that it is '
                                    'currently running.')
```

Comment:
If I'm reading this correctly, this error would still be hit if run somewhere that doesn't have Docker installed and running – is that right?

If so, this limits the upside of this feature; we would no longer require the image upload, but this wouldn't run somewhere like a hosted Jupyter notebook.

Reply:
Sorry for not catching this in the last review! I think it's worth fixing before merging.